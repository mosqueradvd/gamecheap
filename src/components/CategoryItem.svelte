<script>
  export let category;
  export let src;
  export let alt;
  export let href;
</script>

<style>
  .category-item {
    width: 250px;
    height: 60px;
    border-radius: 25px;
    padding: 0 20px;
    margin: 0 auto 40px;
    font-size: 18px;
  }
  .nintendo {
    background: #e50020;
    color: #fff;
    position: relative;
    right: 10vw;
    cursor: pointer;
    transition: all 0.2s ease-in-out;
    box-shadow: 10px 10px 10px 0px rgba(0,0,0,0.28);
  }
  .playstation {
    background: #007aff;
    color: #fff;
    position: relative;
    left: 10vw;
    cursor: pointer;
    transition: all 0.2s ease-in-out;
    box-shadow: 10px 10px 10px 0px rgba(0,0,0,0.28);
  }
  .xbox {
    background: #107c10;
    color: #fff;
    position: relative;
    right: 10vw;
    cursor: pointer;
    transition: all 0.2s ease-in-out;
    box-shadow: 10px 10px 10px 0px rgba(0,0,0,0.28);
  }
  .category-item a {
    text-decoration: none;
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    height: 100%;
  }
  
  .nintendo:hover,
  .playstation:hover,
  .xbox:hover {
    transform: scale(1.1);
  }
  @media screen and (min-width: 700px) {
    .category-item {
      width: 500px;
      height: 120px;
      font-size: 36px;
    }
  }
  @media screen and (min-width: 1000px) {
    .category-item {
      width: 800px;
      height: 150px;
      font-size: 52px;
      margin: 0 auto 20px;
    }
  }
</style>

<div class={`category-item ${category}`}>
  <a href={href}>
    <p style="text-transform: capitalize;">{category}</p>
    <img src={src} alt={alt} style="height: 90%;"/>
  </a>
</div>
