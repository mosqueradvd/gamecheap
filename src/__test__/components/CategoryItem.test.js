import { render as stlRender } from '@testing-library/svelte';
import '@testing-library/jest-dom/extend-expect';
import CategoryItem from '../../components/CategoryItem.svelte';

describe('<CategoryItem /> Component', () => {  
  test('Check the UI of the CategoryItem component', () => {
    const categoryItem = stlRender(CategoryItem)
  
    expect(categoryItem).toMatchSnapshot()
  })
})