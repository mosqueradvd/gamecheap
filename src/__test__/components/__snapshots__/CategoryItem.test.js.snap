// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryItem /> Component Check the UI of the CategoryItem component 1`] = `
Object {
  "component": CategoryItem {
    "$$": Object {
      "after_update": Array [],
      "before_update": Array [],
      "bound": Object {},
      "callbacks": Object {},
      "context": Map {},
      "ctx": Array [
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      "dirty": Array [
        -1,
      ],
      "fragment": Object {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": Array [
        [Function],
      ],
      "on_mount": Array [],
      "props": Object {
        "alt": 2,
        "category": 0,
        "href": 3,
        "src": 1,
      },
      "update": [Function],
    },
    "$capture_state": [Function],
    "$inject_state": [Function],
    "$set": [Function],
  },
  "container": <body>
    <div>
      <div
        class="category-item undefined"
      >
        <a>
          <p
            style="text-transform: capitalize;"
          >
            undefined
          </p>
           
          <img
            style="width: 100px;"
          />
        </a>
      </div>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
